### PR TITLE
Add CNext check for message content intent

### PR DESCRIPTION
**Pull request by VelvetThePanda**
# Summary
As the title says.

# Details
CNext logs an error if the intent config doesn't specify `GuildMessages` or `DirectMessages`, but as of V10, it's required to have the `MessageContents` intent as well.

# Changes proposed
- Add `DiscordIntents.MessageContents`
- Add `DiscordIntents.MessageContents` to `DiscordIntents.All`
- Add `DiscordIntents.MessageContents` to `Utilities.HasMessageIntents`
- Bump GW and API version to 10 (V10).

# Notes
This is a breaking change and should be announced on the server.

### CODE REVIEW

**Comment by OoLunar** [comment]
The bump to V10 should be in a different PR for further review and extreme testing. Something something PR scope - Emzi, probably

**Comment by VelvetThePanda** [comment]
V10 is necessary for the intent which is necessary for the check. V9 is considered deprecated at this time anyways.

File: DSharpPlus/Clients/DiscordClient.WebSocket.cs [hunk 1]
```diff
@@ -137,7 +137,7 @@ internal async Task InternalConnectAsync()
             this._webSocketClient.ExceptionThrown += SocketOnException;
 
             var gwuri = new QueryUriBuilder(this.GatewayUri)
-                .AddParameter("v", "9")
+                .AddParameter("v", "10")
```
Comment (Naamloos) [inline DSharpPlus/Clients/DiscordClient.WebSocket.cs:140]: I wonder why this doesn't use the same CONST var as in Endpoints.cs, as both are the same at all times (we don't mix and match API/GW version anyway)

**Comment by VelvetThePanda** [comment]
"API and Gateway are not in lockstep" - Emzi

File: DSharpPlus/Clients/DiscordClient.WebSocket.cs (hunk 1 above)
Comment (Naamloos) [reply DSharpPlus/Clients/DiscordClient.WebSocket.cs:140]: `Endpoints.API_VERSION`

**Comment by Naamloos** [comment]
> "API and Gateway are not in lockstep" - Emzi

they aren't?

**Comment by VelvetThePanda** [comment]
They should be; I'm just citing Emzi's original reasoning.

**Review by Naamloos — APPROVED** [review]

**Review by ExaInsanity — APPROVED** [review]

**Comment by Naamloos** [comment]
two approvals; merging :)

**Pull request merged**